### ISSUE TITLE
Inclusion BUG: Include relations always reference the modelFrom primary key whether it should be the modelFrom ForeignKey

### ISSUE THREAD

**Issue by FoysalOsmany** [strongloop/loopback-datasource-juggler#619]
@clarkbw @fabien @albertoleal @raymondfeng @bajtos @altsang @ritch 
I am having a serious issue I tried in a lot of ways but could not find it either in documentation or in the codebase.

Suppose I have 2 models

**ModelA**
```
{
    "name": "ModelA",
    "options": {
        "idInjection": false,
        "mssql": {
            "schema": "dbo",
            "table": "ModelA"
        },
        "relations": {
            "medCode": {
                "type": "hasMany",
                "model": "ModelB",
                "foreignKey": "code"
            },
```

**ModelB**
```
{
    "name": "ModelB",
    "options": {
        "idInjection": false,
        "mssql": {
            "schema": "dbo",
            "table": "ModelB"
        }
    },
    "properties": {
        "code": {
            "type": "Number",
            "required": true,
            "length": null,
            "precision": 19,
            "scale": 0,
            "mssql": {
                "columnName": "Code",
                "dataType": "bigint",
                "dataLength": null,
                "dataPrecision": 19,
                "dataScale": 0,
                "nullable": "NO"
            },
            "id": true,
            "generated": false
        },
```

So I am trying to aggregate the ModelB with ModelA as I have the hasMany ModelB with a relation called medCode.

```
ModelA.find({
            include: [
                {
                    relation: 'medCode'
                }
            ]
```

Expected Behaviour should be ModelA aggregate ModelB where medCode match the code which is PK of ModelB.

But Currently the Behaviour is ModelA aggregate ModelB where code = id where id is the PK of ModelA

***IN SHORT***
If the relation is like this 
"relations": {
            "medCode": {
                "type": "hasMany",
                "model": "ModelB",
                "foreignKey": "code"
            },

Then include: "medCode" should map the ModelA.medCode with ModelB.code

But it is doing ModelA.id with ModelB.code where id is the PK of ModelA

**Comment by clarkbw** [comment]
> But it is doing ModelA.id with ModelB.code where id is the PK of ModelA

I'm confused, I don't think is a bug, this is the behavior of the `foreignKey` attribute as far as I can tell. 

Here's some output of a find as you describe above with different model definitions.

With `foreignKey: code`.  Each of the ModelB objects are related to the ModelA however because `code` is the foreignKey the relation can only map to codes that also exist as ids in ModelA.
```
{ name: 'ma1', id: 1, medCode: [ { code: 1 } ] }
{ name: 'ma2', id: 2, medCode: [ { code: 2 } ] }
```

Without the `foreignKey` attribute strongloop creates a new key `modelAId` to tie the two models together and the code value does not have to map to the primary key of ModelA.
```
{ name: 'ma1', id: 1, medCode: [ { code: 3, modelAId: 1 } ] }
{ name: 'ma2', id: 2, medCode: [] }
```

**Comment by loay** [comment]
Hi @FoysalOsmany,
Does @clarkbw  answer your question? Thanks

**Comment by loay** [comment]
I am closing this issue. If you are still experiencing the same thing, please feel free to leave a comment and I will re-open the case. Thanks.
